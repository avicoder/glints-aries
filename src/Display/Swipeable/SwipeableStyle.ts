import styled from 'styled-components';

import { ScreenSize } from '../../Utils/StyleConfig';
import { Greyscale } from '../../Utils/Colors';

export const SwipeableContainer = styled.div`
  display: flex;
  position: relative;
  overflow-x: auto;

  /* make the horizontal scrollbar visible on desktop */
  &::-webkit-scrollbar {
    appearance: none;
    height: 7px;
  }

  /* make the horizontal scrollbar visible on desktop */
  &::-webkit-scrollbar-thumb {
    border-radius: 4px;
    background-color: rgba(0, 0, 0, 0.25);
  }

  @media (max-width: ${ScreenSize.desktopS}px) {
    scrollbar-width: none; /* hide scrollbar on firefox */

    &::-webkit-scrollbar {
      display: none;
    }
  }
`;

export const SwipeableItemWrapper = styled.div`
  display: inline-flex;
  margin: 0 1em;
  padding: 1em 0;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
`;

export const SwipeableCardExample = styled.div`
  width: 150px;
  height: 150px;
  background: ${Greyscale.white};
  box-shadow: 0 10px 15px 0 rgba(0, 0, 0, 0.05);
  display: flex;
  justify-content: center;
  align-items: center;
`;
