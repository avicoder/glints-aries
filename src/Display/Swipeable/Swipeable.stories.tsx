import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Swipeable, Props } from './Swipeable';
import { SwipeableCardExample } from './SwipeableStyle';

export default {
  title: 'Display/Swipeable',
  component: Swipeable,
  argTypes: {
    className: {
      table: {
        disable: true,
      },
    },
  },
} as Meta;

const Template: Story<Props> = ({ ...args }) => {
  return (
    <>
      <Swipeable {...args}>
        {Array.from(Array(10).keys()).map(number => (
          <Swipeable.Item key={number}>
            <SwipeableCardExample>
              <h1>Card. {number}</h1>
            </SwipeableCardExample>
          </Swipeable.Item>
        ))}
      </Swipeable>
    </>
  );
};

export const Default = Template.bind({});
