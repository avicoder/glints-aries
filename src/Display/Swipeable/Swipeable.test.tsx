import * as React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Swipeable from './Swipeable';

describe('<Swipeable /> prop className', () => {
  const matchSnapshot = (className: string) => {
    test(`should match snapshot when class name ${className} is passed`, () => {
      const { asFragment } = render(
        <Swipeable className={className}>
          <Swipeable.Item className={className}>Item</Swipeable.Item>
        </Swipeable>
      );
      // validate class name from snapshot
      expect(asFragment()).toMatchSnapshot();
    });
  };

  ['test', undefined].forEach(matchSnapshot);
});

describe('<Swipeable /> prop children', () => {
  const itemCount = 10;
  test(`should render ${itemCount} <Swipeable.Item />`, () => {
    render(
      <Swipeable>
        {Array.from(Array(itemCount).keys()).map(number => (
          <Swipeable.Item key={number}>Card. {number}</Swipeable.Item>
        ))}
      </Swipeable>
    );

    const items = document.querySelectorAll('.swipeable-itemwrapper');
    expect(items).toHaveLength(itemCount);
  });
});

test('<Swipeable /> should be scrollable', () => {
  render(
    <Swipeable>
      <Swipeable.Item>Item</Swipeable.Item>
    </Swipeable>
  );

  const swipeable = document.querySelector('.aries-swipeable');
  expect(swipeable).toHaveStyle('overflow-x: auto');
});
