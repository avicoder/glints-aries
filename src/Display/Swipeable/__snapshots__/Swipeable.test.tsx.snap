// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Swipeable /> prop className should match snapshot when class name test is passed 1`] = `
<DocumentFragment>
  <div
    class="SwipeableStyle__SwipeableContainer-xvh6jb-0 ckHALM aries-swipeable test"
  >
    <div
      class="SwipeableStyle__SwipeableItemWrapper-xvh6jb-1 cIchyd swipeable-itemwrapper test"
    >
      Item
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Swipeable /> prop className should match snapshot when class name undefined is passed 1`] = `
<DocumentFragment>
  <div
    class="SwipeableStyle__SwipeableContainer-xvh6jb-0 ckHALM aries-swipeable"
  >
    <div
      class="SwipeableStyle__SwipeableItemWrapper-xvh6jb-1 cIchyd swipeable-itemwrapper"
    >
      Item
    </div>
  </div>
</DocumentFragment>
`;
